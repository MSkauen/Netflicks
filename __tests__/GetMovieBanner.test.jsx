import React from "react";
import ReactDOM from "react-dom";
import { describe, expect, it } from "@jest/globals";
import { act } from "react-dom/test-utils";
import GetRandomMovieBanner from "../src/client/components/MovieBanner";
import {MemoryRouter} from "react-router";

async function render(component) {
    const container = document.createElement("div");
    document.body.appendChild(container);
    await act(async () => {
        ReactDOM.render(<MemoryRouter>{component}</MemoryRouter>, container);
    });
    return container;
}

describe("get movie banner", () => {
    it("show random movie banner", async () => {
        const movieApi = {
            async getRandomMovie() {
                return (
                    {
                        id: 10,
                        name: "Forest movie",
                        genres: ["Drama", "Action"],
                        rating: {
                            average: 6.4
                        },
                        summary: "Great movie about forests."
                    }
                );
            },
            getMovieBackground: async () => [
                {
                    id: 1,
                    resolutions: {
                        original: {
                            url: "https://static.tvmaze.com/uploads/images/medium_portrait/0/1.jpg"
                        }
                    },
                    type: "background"
                },
            ]
        };

        const container = await render(<GetRandomMovieBanner movieApi={ movieApi } />);
        expect(container.innerHTML).toMatchSnapshot();
        expect(container.querySelector("img").src).toEqual(
            "https://static.tvmaze.com/uploads/images/medium_portrait/0/1.jpg"
        );
    });
})