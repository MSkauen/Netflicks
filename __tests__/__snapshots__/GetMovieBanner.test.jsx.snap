// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get movie banner show random movie banner 1`] = `"<header class="banner"><div class="banner_contents"><img class="movie_banner" src="https://static.tvmaze.com/uploads/images/medium_portrait/0/1.jpg"><h1 class="banner_title">Forest movie</h1><div class="banner_buttons"><button class="banner_button">Play</button><button class="banner_button">Add to My list</button></div><h1 class="banner_description">Great movie about forests.</h1></div><div class="banner_fadeBottom"></div></header>"`;
